src/auspex/experiment.py

```diff
@@ -421,60 +434,16 @@ def run_sweeps(self):
         if len(self.plotters) > 0:
             logger.debug("Found %d plotters", len(self.plotters))
 
-            from .plotting import BokehServerProcess
-            from bokeh.client import push_session
-            from bokeh.layouts import row, gridplot
-            from bokeh.io import curdoc
-            from bokeh.models.widgets import Panel, Tabs
-
-            # If anybody has requested notebook plots, show them all in notebook
-            run_in_notebook = True in [p.run_in_notebook for p in self.plotters]
+            from .plotting import MatplotServerThread
 
-            bokeh_process = BokehServerProcess(notebook=run_in_notebook)
-            bokeh_process.run()
-
-            tabs = True #not run_in_notebook # Tabs seem a bit sluggish in jupyter notebooks...
-            if tabs:
-                container = Tabs(tabs=[Panel(child=p.fig, title=p.name) for p in self.plotters])
-            else:
-                if len(self.plotters) <= 2:
-                    container = row(*[p.fig for p in self.plotters])
-                else:
-                    padded_list = self.plotters[:]
-                    if len(padded_list)%2 != 0:
-                        padded_list.append(None)
-                    grid = list(zip(*[iter(padded_list)]*2))
-                    container = gridplot(grid)
-
-            doc = curdoc()
-            doc.clear()
-            doc.add_root(container)
-            session = push_session(doc)
-
-            for p in self.plotters:
-                p.session = session
-
-            if run_in_notebook:
-                logger.debug("Displaying in iPython notebook")
-                from bokeh.embed import autoload_server, components
-                from bokeh.io import output_notebook
-                from IPython.display import display, HTML, clear_output
-
-                # clear_output()
-                output_notebook()
-                script = autoload_server(model=None, session_id=session.id)
-                html = \
-                        """
-                        <html>
-                        <head></head>
-                        <body>
-                        {}
-                        </body>
-                        </html>
-                        """.format(script)
-                display(HTML(html))
-            else:
-                session.show(container)
+            plot_desc = {p.name: p.desc() for p in self.plotters}
+            self.plot_server = MatplotServerThread(plot_desc)
+            for plotter in self.plotters:
+                plotter.plot_server = self.plot_server
+            time.sleep(0.5)
+            client_path = os.path.join(os.path.dirname(os.path.abspath(__file__)),"matplotlib-client.py")
+            subprocess.Popen(['python', client_path, 'localhost'], env=os.environ.copy())
+            time.sleep(1)
```

Comment:
Everything runs fine without the sleep, but for short experiments the data can come in before the plotter has actually launched and connected, thus the silly delay. I think IPC is a good idea at some point.